polars/polars-core/src/fmt.rs

```diff
@@ -359,7 +359,10 @@ impl Display for DataFrame {
             let field_to_str = |f: &Field| {
                 let name = make_str_val(f.name(), str_truncate);
                 let lower_bounds = std::cmp::max(5, std::cmp::min(12, name.len()));
-                let s = format!("{}\n---\n{}", name, f.data_type());
+                let mut s = format!("{}\n---\n{}", name, f.data_type());
```

Comment:
This env var is checked on 3 location and every time we set a field value. Can you set a boolean flag on top of this function?